### PR TITLE
Add Nokogiri XML builders (Incomplete: request for review)

### PR DESCRIPTION
**Pull request by llimllib**
I wanted to ensure that the XML returned from the erb and the nokogiri methods was the same, so for now I left both in and wrote tests to verify that it is. If/when this PR gets close to approval, I'll pull the erb code and the tests for equivalence.

Some Qs:

* Is it worth all requests inheriting from a common base class to get the namespaces?
* I didn't find a good way of composing Nokogiri XML builders, but that doesn't mean one doesn't exist; let me know if you can figure out how
* Should each request go in its own file?

:eyes: @alex @amoose

this is an attempt to implement #8

### CODE REVIEW

**Comment by alex** [comment]
1. I would like to find a way to do this without inheritance (I'm not a fan of subclassing); I think most of the use cases we want can be done with functions.
2. I _think_ you can just pass them around between functions? e.g. I think we could do something like

```ruby
soap_builder do |xml|
    xml['v4'].getDocumentTypes
end
```

3. Once we have this `soap_buidler` function it should be sufficiently short for many of them that doing them inline is ok?

4. You have merge conflicts :-(

**Comment by llimllib** [comment]
734f4d2 should fix 1,2, and 3. I'm not worried about the merge conflicts, a rebase will be easy and I still need to delete the erb functions once we're ready to go anyway.

**Comment by alex** [comment]
Haven't reviewed the tests or other bits; but the builders themselves look all right to me.

File: src/vbms/requests.rb [hunk 1]
```diff
@@ -42,6 +59,38 @@ def render_xml()
         return self.template.result(binding)
       end
 
+      def received_date
+        @received_at.getlocal("-05:00").strftime("%Y-%m-%d-05:00")
```
Comment (amoose) [inline src/vbms/requests.rb]: Could this cause issues Mar 8-Nov 1?

Comment (amoose) [inline src/vbms/requests.rb]: Since DC will soon be moving to EDT instead of EST, should we default to EST but inherit from ENV var for compatibility?

Comment (llimllib) [inline src/vbms/requests.rb]: oh goodness... @greggersh discovered that we needed to set the time to -0500, I had been sending UTC, but I'm not sure if we have to be on EST or EDT. Any ideas Greg?

(The only sane thing to do is of course to send UTC, but that was specifically verboten by VBMS)

Comment (greggersh) [inline src/vbms/requests.rb]: Good grief, I do not at all remember what the deal was there. I vaguely recall it was based on some artifacts from the sample XML files.

This has been running fine for longer than we've been in EDT, so I'd recommend leaving as is.

Comment (amoose) [inline src/vbms/requests.rb]: :hear_no_evil: :see_no_evil: :speak_no_evil:

**Comment by amoose** [comment]
This looks like a great start, @llimllib. I would prefer the values for name to be class constants. Also, let's break request specs out in to subfolder to match ~~lib~~ src structure. Lastly, rubo:cop: is going to have a few things to say about this PR :grin:

**Comment by llimllib** [comment]
> I would prefer the values for name to be class constants

is https://github.com/department-of-veterans-affairs/connect_vbms/commit/8661aa14dfd07c79a354831fa01f0361df005363 what you were thinking?

>  Lastly, rubo:cop: is going to have a few things to say about this PR

Yup, the rebase got a bit less fun this afternoon :)

**Comment by llimllib** [comment]
closing, superceded by #28

**Pull request closed without merging**